### PR TITLE
Week 2

### PR DESCRIPTION
**Pull request by evajaum**
All the models and answers for week 2!

### CODE REVIEW

File: greenery/models/marts/core/fact_orders.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+         FROM   {{ ref('order_items') }})
+SELECT orders.order_id,
+       orders.created_at,
+       orders.status,
+       orders.order_total,
+       orders.order_cost,
+       orders.shipping_cost,
+       Round(Cast(orders.order_total - orders.order_cost - orders.shipping_cost
+                  AS
+                  NUMERIC), 2)                             AS order_profit,
```
Comment (zoee-liang) [inline greenery/models/marts/core/fact_orders.sql:20]: I think you profit formula is correct (at least based on the information we currently have)... So maybe that means unfortunately our Greenery really does not make profits :sob:

File: greenery/models/marts/core/fact_orders.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        tests:
+          - not_null
+          - unique
+      - name: created_at
+        description: Creation of order.
+      - name: status
+        description: status
+        tests:
+          - accepted_values:
+              values: ['shipped', 'preparing', 'delivered']
```
Comment (zoee-liang) [inline greenery/models/marts/core/fact_orders.yml:18]: I like this test!

File: greenery/models/marts/core/dim_users.sql [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+         FROM   events
+         GROUP  BY 1)
+SELECT users.user_id,
+       users.created_at,
+       users.updated_at,
+       order_stats.total_orders,
+       order_stats.first_order_created_at,
+       order_stats.latest_order_created_at,
+       life_time.total_products,
+       Round(Cast(life_time.total_value AS NUMERIC), 2) AS lifetime_value_usd,
```
Comment (zoee-liang) [inline greenery/models/marts/core/dim_users.sql:46]: This is a great metric!

File: greenery/models/marts/core/fact_orders.sql [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+       orders.delivered_at - orders.created_at             AS delivery_time,
+       orders.delivered_at <= orders.estimated_delivery_at AS is_on_time,
+       order_items.distinct_products,
+       order_items.total_products,
```
Comment (zoee-liang) [inline greenery/models/marts/core/fact_orders.sql:24]: This is probably very minor and maybe even personal, but as an analyst I feel like the column names `distinct_products` and `total_products` are a little bit confusing if we look at them together. Would something like `total_sold_items` makes sense? (just suggesting, definitely feel free to disagree!)

File: greenery/tests/profit_is_positive.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+-- We should make some profit with each order!
+select
+    order_id,
+    sum(order_profit)
+from {{ ref('fact_orders' )}}
+group by 1 
+having (sum(order_profit) <= 0)
```
Comment (zoee-liang) [inline greenery/tests/profit_is_positive.sql:7]: Wow, really great idea!

File: greenery/weekly_questions/2_week.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+* device 
+* country / city
+* experience with plants :D 
+
+### Explain the marts models you added. Why did you organize the models in the way you did?
+#### Core
+Information per user and product in the corresponding dim tables and per order in the fact table.
+An intermediate order_items table with a summary of number of products per order e.g.
+As the core models will be available to everybody in the company, I didn't want to include any private information
+in the tables, such as clear names, emails or addresses of our users.
```
Comment (zoee-liang) [inline greenery/weekly_questions/2_week.md:43]: That's a great callout!

File: greenery/weekly_questions/2_week.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+* higher number of page interactions
+and users are less likely with the opposite.
+
+More data could be helpful:
+* demographics
+* referral codes
+* acquisition channel
+* device 
+* country / city
+* experience with plants :D
```
Comment (zoee-liang) [inline greenery/weekly_questions/2_week.md:36]: I really love your answer of this question! I can tell you're definitely a data person :)

File: greenery/models/marts/core/dim_products.sql [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+                       ON oi.order_id = o.order_id
+         GROUP  BY 1)
+SELECT 
+products.product_id,
+products.name,
+products.price as price_usd,
+products.inventory,
+order_stats.total_ordered,
+order_stats.first_ordered_at,
+order_stats.last_ordered_at
```
Comment (zoee-liang) [inline greenery/models/marts/core/dim_products.sql:22]: :+1:

**Review by zoee-liang — APPROVED** [review]
Eva - You definitely did a fantastic job! I've actually learned a LOT from your project myself. So thank you, for having me as one of your reviewers! :)

File: greenery/models/marts/core/fact_orders.sql (hunk 4 above)
Comment (evajaum) [reply greenery/models/marts/core/fact_orders.sql]: Yes! Thanks for the suggestion, I like it more. Naming is hard!

File: greenery/weekly_questions/2_week.md (hunk 7 above)
Comment (evajaum) [reply greenery/weekly_questions/2_week.md:36]: Aaaaw, thank you so much for all your positive comments! Made a nice start in the day for me and is really encouraging!

**Pull request merged**